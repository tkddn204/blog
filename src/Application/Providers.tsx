import React, { FC } from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { Global, css } from '@emotion/core'
import { ThemeProvider } from 'emotion-theming'
import emotionNormalize from 'emotion-normalize'

import firebase from 'firebase/app'
import 'firebase/auth'
import 'firebase/firestore'
import 'firebase/storage'
import { ReactReduxFirebaseProvider } from 'react-redux-firebase'
import { createFirestoreInstance } from 'redux-firestore'
import { PersistGate } from 'redux-persist/integration/react'
import { BrowserRouter } from 'react-router-dom'
import { firebaseConfig, reduxFirebaseConfig } from './FirebaseConfig'
import { persistor, store } from './Store'

import './I18nConfig'
import LocaleProvider from '../Features/locale/LocaleProvider'
import { commonTheme } from './Theme'
// import SelectThemeProvider from '../Features/theme/SelectThemeProvider'

// Initialize Firebase
firebase.initializeApp(firebaseConfig)
firebase.auth()
firebase.storage()
const firestore = firebase.firestore()
if (window.location.hostname === 'localhost') {
  firestore.settings({
    host: 'localhost:5002',
    ssl: false,
  })
}

const Providers: FC = ({ children }) => (
  <React.StrictMode>
    <ReduxProvider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <ReactReduxFirebaseProvider
          firebase={firebase}
          config={reduxFirebaseConfig}
          dispatch={store.dispatch}
          createFirestoreInstance={createFirestoreInstance}
        >
          <BrowserRouter>
            <ThemeProvider theme={commonTheme}>
              <LocaleProvider>
                <Global
                  styles={css`
                    ${emotionNormalize}
                  `}
                />
                {children}
              </LocaleProvider>
            </ThemeProvider>
          </BrowserRouter>
        </ReactReduxFirebaseProvider>
      </PersistGate>
    </ReduxProvider>
  </React.StrictMode>
)

export default Providers
