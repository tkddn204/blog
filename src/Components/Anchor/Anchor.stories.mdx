import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks'
import Anchor from '.'

<Meta
  title="Components/Anchor"
  component={Anchor}
  argTypes={{
    color: { control: 'color' },
    text: { control: 'text' }
  }}
/>

# 🍎 Anchor

앵커입니다. 앵커는 `<a>` 태그 컴포넌트입니다.

## 예제

export const Template = (args) => <Anchor {...args}>{args.text}</Anchor>

<Canvas>
  <Story name="anchor" args={{ text: "앵커 - anchor" }}>
    {Template.bind()}
  </Story>
</Canvas>

## Props

|속성|타입|설명|
|:--|:--|:--|
|color|string|앵커의 색을 지정합니다. 올바른 색상 문자열이 아닐 경우 기본 앵커 색을 따릅니다.|
