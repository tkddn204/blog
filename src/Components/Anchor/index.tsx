/** @jsx jsx */
import { FCEP, HTMLProps } from 'react'
import { Link } from 'react-router-dom'
import { css, jsx } from '@emotion/core'
import useStyle from '../../Hooks/useStyle'
import { ThemeType } from '../../Types/theme'

const style = (theme: ThemeType) => css`
  color: ${theme.color.pink['5']};

  &:hover {
    text-decoration: underline;
  }
`

const darkStyle = (theme: ThemeType) => css`
  color: ${theme.color.pink['2']};
`

interface Props extends HTMLProps<HTMLAnchorElement> {
  link?: string
}

const Anchor: FCEP<Props> = ({ link, children, className }) => (
  <Link to={link || '/'} css={useStyle(style, darkStyle)} className={className}>
    {children}
  </Link>
)

export default Anchor
