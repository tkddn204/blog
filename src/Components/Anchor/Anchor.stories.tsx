import React, { ReactNode } from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'
import Anchor from '.'
import DocTemplate from '../../StorybookDocTemplate'
import { LightTheme } from '../../Application/Theme'

export default {
  title: 'Anchor',
  component: Anchor,
  argTypes: {
    children: {
      description: '앵커 안에 들어가는 텍스트',
      type: {
        required: true,
      },
      table: {
        type: {
          summary: 'string',
        },
        defaultValue: {
          summary: '-',
        },
      },
      defaultValue: '기본 앵커',
      control: 'text',
    },
    color: {
      description:
        '앵커의 색을 지정합니다. 올바른 색상 문자열이 아닐 경우 테마 앵커의 색상을 따릅니다.',
      defaultValue: LightTheme.colors.anchor,
      table: {
        type: {
          summary: 'color',
        },
        defaultValue: {
          summary: LightTheme.colors.anchor,
        },
      },
      control: 'color',
    },
  },
  parameters: {
    docs: {
      page: () =>
        DocTemplate({
          title: '🍎 Anchor',
          description: '앵커입니다. 앵커는 `<a>` 태그 컴포넌트입니다.',
        }),
    },
  },
} as Meta

interface StoryAnchorProps {
  color: string
  children: ReactNode
}
const anchorTemplate: Story<StoryAnchorProps> = (
  args,
  { globals: { theme, locale } }
) => {
  const { color } = args
  const text = locale !== 'ko' ? 'Anchor' : args.children

  return theme !== 'notUsed' ? (
    <Anchor>{text}</Anchor>
  ) : (
    <Anchor theme={{ colors: { anchor: color } }}>{text}</Anchor>
  )
}

export const DefaultAnchor = anchorTemplate
