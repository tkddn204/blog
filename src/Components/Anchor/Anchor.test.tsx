import React from 'react'
import { render } from '@testing-library/react'

import Anchor from '.'
import Providers from '../../Application/Providers'

test('renders correctly when there are some texts', () => {
  const someText = '텍스트'
  const { getByText, container } = render(
    <Providers>
      <Anchor>{someText}</Anchor>
    </Providers>
  )

  expect(getByText(someText).textContent).toEqual(someText)
  expect(container).toMatchSnapshot()
})
