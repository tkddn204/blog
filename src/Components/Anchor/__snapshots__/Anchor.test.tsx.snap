// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are some texts 1`] = `
.emotion-0 {
  color: #f06595;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <a
    class="emotion-0"
    href="/"
  >
    텍스트
  </a>
</div>
`;
